RecognizeDtmf/RecognizeDtmf.cs

```diff
@@ -0,0 +1,341 @@
+
+                Logger.LogMessage(Logger.MessageType.INFORMATION, "Performing CreateCall operation");
+
+                var call = await callClient.CreateCallAsync(createCallOption).ConfigureAwait(false);
+
+                Logger.LogMessage(Logger.MessageType.INFORMATION, $"CreateCallConnectionAsync response --> {call.GetRawResponse()}, Call Connection Id: { call.Value.CallConnectionProperties.CallConnectionId}");
+
+                Logger.LogMessage(Logger.MessageType.INFORMATION, $"Call initiated with Call Connection id: { call.Value.CallConnectionProperties.CallConnectionId}");
+
+                RegisterToCallStateChangeEvent(call.Value.CallConnectionProperties.CallConnectionId);
+
+                //Wait for operation to complete
+                await callEstablishedTask.Task.ConfigureAwait(false);
+
+                return call.Value.CallConnection;
+            }
+            catch (Exception ex)
+            {
+                Logger.LogMessage(Logger.MessageType.ERROR, string.Format("Failure occured while creating/establishing the call. Exception: {0}", ex.Message));
+                throw ex;
+            }
+        }
+
+        private async Task StartRecognizingDtmf(string targetPhoneNumber)
+        {
+            if (reportCancellationToken.IsCancellationRequested)
+            {
+                Logger.LogMessage(Logger.MessageType.INFORMATION, "Cancellation request, PlayAudio will not be performed");
+                return;
+            }
+
+            try
+            {
+                string audioFilePath = callConfiguration.AudioFileUrl + callConfiguration.AudioFileName;
+                PlaySource audioFileUri = new FileSource(new Uri(audioFilePath));
+
+                // listen to play audio events
+                RegisterToPlayAudioResultEvent(callConnection.CallConnectionId);
+
+                //Start recognizing Dtmf Tone
+                var recognizeOptions = new CallMediaRecognizeDtmfOptions(new PhoneNumberIdentifier(targetPhoneNumber), 1);
+                recognizeOptions.InterToneTimeout = TimeSpan.FromSeconds(5);
+                recognizeOptions.InitialSilenceTimeout = TimeSpan.FromSeconds(30);
+                recognizeOptions.InterruptPrompt = true;
+                recognizeOptions.InterruptCallMediaOperation = true;
+                recognizeOptions.Prompt = audioFileUri;
+                recognizeOptions.StopTones = new List<DtmfTone> { DtmfTone.Pound };
+
+                var resp = await callConnection.GetCallMedia().StartRecognizingAsync(recognizeOptions, reportCancellationToken);
+
+                Logger.LogMessage(Logger.MessageType.INFORMATION, $"StartRecognizingAsync response --> " +
+                $"{resp}, Id: {resp.ClientRequestId}, Status: {resp.Status}");
+
+                //Wait for 30 secs for input
+                var completedTask = await Task.WhenAny(playAudioCompletedTask.Task, Task.Delay(30 * 1000)).ConfigureAwait(false);
```

Comment:
This is not needed if we await Recognize event and as we discussed before this should be recognizeCompletedTask and not playAudioCompleted task